backend/src/main/java/com/bakdata/conquery/sql/conversion/dialect/HanaSqlFunctionProvider.java

```diff
@@ -0,0 +1,138 @@
+package com.bakdata.conquery.sql.conversion.dialect;
+
+import java.sql.Date;
+import java.time.temporal.ChronoUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.bakdata.conquery.models.common.daterange.CDateRange;
+import com.bakdata.conquery.models.datasets.Column;
+import com.bakdata.conquery.models.datasets.concepts.ValidityDate;
+import com.bakdata.conquery.sql.models.ColumnDateRange;
+import org.jooq.Condition;
+import org.jooq.Field;
+import org.jooq.Name;
+import org.jooq.impl.DSL;
+
+public class HanaSqlFunctionProvider implements SqlFunctionProvider {
+
+	private static final String INFINITY_DATE_VALUE = "9999-12-31";
+	private static final String MINUS_INFINITY_DATE_VALUE = "0001-01-01";
+
+	@Override
+	public Condition dateRestriction(ColumnDateRange dateRestriction, ColumnDateRange validityDate) {
+
+		if (dateRestriction.isSingleColumnRange() || validityDate.isSingleColumnRange()) {
+			throw new UnsupportedOperationException("HANA does not support single column ranges.");
+		}
+
+		Condition dateRestrictionStartsBeforeDate = dateRestriction.getStart().lessOrEqual(validityDate.getEnd());
+		Condition dateRestrictionEndsAfterDate = dateRestriction.getEnd().greaterOrEqual(validityDate.getStart());
+
+		return DSL.condition(dateRestrictionStartsBeforeDate.and(dateRestrictionEndsAfterDate));
+	}
+
+	@Override
+	public ColumnDateRange daterange(CDateRange dateRestriction) {
+
+		String startDateExpression = MINUS_INFINITY_DATE_VALUE;
+		String endDateExpression = INFINITY_DATE_VALUE;
+
+		if (dateRestriction.hasLowerBound()) {
+			startDateExpression = dateRestriction.getMin().toString();
+		}
+		if (dateRestriction.hasUpperBound()) {
+			endDateExpression = dateRestriction.getMax().toString();
+		}
+
+		return ColumnDateRange.of(toDate(startDateExpression), toDate(endDateExpression))
```

Comment:
```suggestion
		LocalDate startDateExpression = LocalDate.parse(MINUS_INFINITY_DATE_VALUE); // oder so, aber als static variable
		LocalDate endDateExpression = INFINITY_DATE_VALUE;

		if (dateRestriction.hasLowerBound()) {
			startDateExpression = dateRestriction.getMin();
		}
		if (dateRestriction.hasUpperBound()) {
			endDateExpression = dateRestriction.getMax();
		}

		return ColumnDateRange.ofstartDateExpression, endDateExpression)
```

Reply:
Jetzt verstehe ich. Allerdings wandelt die `toDate()` Methode ja in ein `Field<Date>` und muss dann zwangsläufig noch aufgerufen werden. Und da ich das noch an anderen Stellen brauche, wo allerdings ein Date und kein LocalDate vorliegt, schien es mir am einfachsten den Parameter von `toDate()` als String zu haben.